packages/app/lib/src/features/home/application/uploaded_image_service.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/foundation.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+@freezed
+class UploadedImage {
+  // Might need to add IsWeb in here
+  UploadedImage({
+    required this.imageId, //figure out how to do image ids
+    //currently working on multi image upload support. Will need to figure out how to store multiple images. Should just make appwrite imageIds an array of strings and store all ids and loop the upload function
+    required this.file,
+  });
+  String imageId;
+  PlatformFile file;
+}
+
+@riverpod
+class UploadedImagesNotifier extends ChangeNotifier {
+  final uploadedImages = <UploadedImage>[];
+
+  void addImage(UploadedImage image) {
+    uploadedImages.add(image);
+    notifyListeners();
+  }
+
+  void removeImage(String imageId) {
+    // Again, our state is immutable. So we're making a new list instead of
+    // changing the existing list.
+    uploadedImages.remove(
+      uploadedImages.firstWhere((element) => element.imageId == imageId),
+    );
+  }
+}
+
+final uploadImagesProvider =
+    ChangeNotifierProvider<UploadedImagesNotifier>((ref) {
+  return UploadedImagesNotifier();
+});
```

Comment:
```suggestion
@riverpod
class UploadedImageService extends _$UploadedImageService {
  IList<UploadedImage> build() {
    return const IList.empty();
  }

  void addImage(UploadedImage image) {
    state = state.add(image);
  }

  void removeImage(String imageId) {
    // Again, our state is immutable. So we're making a new list instead of
    // changing the existing list.
    state = state.remove(
      uploadedImages.firstWhere((element) => element.imageId == imageId),
    );
  }
}
```

You'll also need to import `package:fast_immutable_collections/fast_immutable_collections.dart` and add part directives for `uploaded_image_service.freezed.dart` & `uploaded_image_service.g.dart`.

Reply:
Were you able to get this run? It's not generating the freed and g files for me. I'm probably missing something though